src/evd/cyclic/evd_cost.cpp

```diff
@@ -172,25 +171,24 @@ size_t blocked_cost_with_subprocedure_evd(size_t n, size_t b, size_t n_iter, siz
     size_t single_mult_block_mul = b * b * b;
     size_t single_add = b * b;
 
-    size_t block_adds = loops * (4 * n_blocks * (12 * single_mult_block_add + 6 * single_add));
-    size_t block_muls = loops * (4 * n_blocks * (12 * single_mult_block_mul));
-    size_t individual_block_ops = loops * subprocedure_cost(4 * b, individual_block_iter);  // 4 individual blocks
+    size_t block_adds = loops * (3 * n_blocks * (2 * single_add + 4 * single_mult_block_add));
+    size_t block_muls = loops * (3 * n_blocks * (4 * single_mult_block_mul));
+    size_t individual_block_ops = loops * base_cost_evd(2 * b, individual_block_iter);  // 4 individual blocks
```

Comment:
Here, I have used `subprocedure_cost` because it is different when compared to `base_cost_evd`. The subprocedure calls the `nevd` function and it is not the same as origial EVD computation. Moreover, both `Ablockmat` and `Vblockmat` in the call `evd_subprocedure_vectorized(Ablockmat, Vblockmat);` are 2b * 2b => 4 * (b*b) in size. Hence I use `subprocedure_cost(4 * b, individual_block_iter);`

Reply:
But why do you use ```nevd``` and not what you used in the original evd computation? Now all evd versions use nevd?

Reply:
I am also a bit unsure about the nevd. Another issue is that the subprocedure here is tolerance based, but the cost function takes the number of iterations as input. I cannot understand how the cost function is correct then.

Reply:
> I am also a bit unsure about the nevd. Another issue is that the subprocedure here is tolerance based, but the cost function takes the number of iterations as input. I cannot understand how the cost function is correct then.

The tol can be removed and be reinstated with our original epoch version of EVD. But `nevd` doesn't actually interfere in the process. I used it to make it kind of uniform when compared to its SVD counterpart

Reply:
But do we want to compare EVD with SVD in the end? I think it would be better to compare with Eigen EVD and also all versions of EVD should use either nevd or what was originally used. So can you guys decide and fix the codes accordingly?

Reply:
> But do we want to compare EVD with SVD in the end? I think it would be better to compare with Eigen EVD and also all versions of EVD should use either nevd or what was originally used. So can you guys decide and fix the codes accordingly?

We don't compare SVD and EVD though. I have removed tolerance now in this PR: https://github.com/ktrianta/jacobi-svd-evd/pull/63

I can remove `nevd` too if you suggest.